### ISSUE TITLE
Add color term to WCS classes

### ISSUE THREAD

**Issue by rmjarvis** [GalSim-developers/GalSim#865]
@gbernstein has astrometric solutions for DES that include a color term.  I'm working on wrapping his solutions into GalSim WCS classes, but we don't currently have any way to deal with the color dependence in GalSim.

I'm proposing to add an optional color term to the WCS classes that would be ignored for all the current WCS classes, but which would provide an appropriate hook to work with Gary's solutions.  Mostly, I just need to have a `color` object that defaults to `None`, which would be passed through the front end methods like `toWorld` down into the implementation methods like `_radec`.

Basically, you would be able to do
```
sky_pos = wcs.toWorld(image_pos, color=gminusi)
```
and if the wcs class supports it, it would use the color value appropriately.

Gary's code uses float values of g-i for the color, but I think since we are just passing along the value to the back-end WCS class, we don't need to enforce that anywhere.  So some other WCS class might use a more complicated "color" object that encapsulates the information in whatever way it needs.

**Comment by rmjarvis** [comment]
Merged via #872
